backend/src/LibExecution/ProgramTypes.fs

```diff
@@ -152,21 +165,68 @@ type Expr =
   | ETuple of id * Expr * Expr * List<Expr>
   | ERecord of id * List<string * Expr>
   | EPipe of id * Expr * Expr * List<Expr>
+
   // Constructors include `Just`, `Nothing`, `Error`, `Ok`.  In practice the
   // expr list is currently always length 1 (for `Just`, `Error`, and `Ok`)
   // or length 0 (for `Nothing`).
+  // TODO: migrate usages of this to usages of EDefinedEnum(FQTypeName.T, ...
   | EConstructor of id * string * List<Expr>
-  | EMatch of id * Expr * List<MatchPattern * Expr>
+
+  /// Supports `match` expressions
+  /// ```fsharp
+  /// match x + 2 with // arg
+  /// // cases
+  /// | pattern -> expr
+  /// | pattern -> expr
+  /// | ...
+  /// ```
+  | EMatch of id * arg : Expr * cases : List<MatchPattern * Expr>
+
   // Placeholder that indicates the target of the Thread. May be movable at
   // some point
   | EPipeTarget of id
-  // EFeatureFlag: id, flagName, condExpr, caseAExpr, caseBExpr
-  | EFeatureFlag of id * string * Expr * Expr * Expr
+
+  /// Like an if statement, but with a label
+  /// TODO: continue describing
+  | EFeatureFlag of
+    id *
+    flagName : string *
+    cond : Expr *
+    caseA : Expr *
+    caseB : Expr
+
+
+  // TODO:
+  // - define EUser
```

Comment:
I wouldn't use EUser here - it seems like there should be ERecord, EConstructor and EFnCall, which each have a FQName also FQName.User, FQName.Package, and FQName.Stdlib.